disk/disk.h

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2023 Pantacor Ltd.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifndef PV_DISK_H
+#define PV_DISK_H
+
+#include "utils/list.h"
+#include "state.h"
+
+typedef enum {
+	DISK_UNKNOWN,
+	DISK_DIR,
+	DISK_DM_CRYPT_VERSATILE,
+	DISK_DM_CRYPT_CAAM,
+	DISK_DM_CRYPT_DCP,
+	DISK_SWAP,
+	DISK_VOLUME
+} pv_disk_t;
+
+typedef enum {
+	DISK_FORMAT_UNKNOWN,
+	DISK_FORMAT_SWAP,
+	DISK_FORMAT_EXT4,
+	DISK_FORMAT_EXT3
+} pv_disk_format_t;
+
+typedef enum {
+	DISK_STATUS_ERROR,
+	DISK_STATUS_MOUNTED,
+	DISK_STATUS_NOT_MOUNTED
+} pv_disk_status_t;
```

Comment:
we can move this to .c if we are not going to use it externally

Reply:
I will check it and move it if it's ok. Thanks!

Reply:
This is used in several places like disk_impl.h and each disk implementation